scripts/website.js

```diff
@@ -76,6 +77,48 @@ const tests = [
   ...acquit.parse(fs.readFileSync(path.join(testPath, 'docs/schemas.test.js')).toString())
 ];
 
+function deleteAllHtmlFiles() {
+  try {
+    fs.unlinkSync('./index.html');
+  } catch (err) {
+    if (err.code !== 'ENOENT') {
+      throw err;
+    }
+  }
+  const foldersToClean = [
+    './docs',
+    './docs/tutorials',
+    './docs/typescript',
+    './docs/api',
+    './docs/source/_docs',
+    './tmp'
+  ];
+  for (const folder of foldersToClean) {
+    let files = [];
+    try {
+      files = fs.readdirSync(folder);
+    } catch (err) {
+      if (err.code === 'ENOENT') {
+        continue;
+      }
+    }
+    for (const file of files) {
+      if (file.endsWith('.html')) {
+        fs.unlinkSync(`${folder}/${file}`);
+      }
+    }
+  }
+}
+
+function moveDocsToTemp() {
+  fs.rmSync('./tmp', { recursive: true });
+  const folder = versionObj.versionedPath.replace(/^\//, '');
+  const directory = fs.readdirSync(folder);
+  for (const file of directory) {
+    fsextra.moveSync(`${folder}/${file}`, `./tmp/${file}`);
+  }
```

Comment:
this likely will need a check for a non-empty `versionObj.versionedPath` or a note in the JSDOC, because if `versionedDeploy` is `false` OR `currentVersion === latestVersion` the `versionedPath` is ""(empty, project root), not `/docs/X.X/` and so would move *all* the project into tmp (and i dont think this is intended, guessing by [`docs:prepare:publish:stable`](https://github.com/Automattic/mongoose/blob/4b065f2aed1391dc9b0b5f5f914d7f01847a3723/package.json#L96C6-L96C33) and the old [`docs:prepare:publish:7x`](https://github.com/Automattic/mongoose/blob/ea855148c7ecdcfc60f46f286165617380083bc7/package.json#L99))
[see this code](https://github.com/Automattic/mongoose/blob/4b065f2aed1391dc9b0b5f5f914d7f01847a3723/scripts/website.js#L261C102-L263)

Reply:
This shouldn't be strictly necessary because `moveDocsToTemp()` is only called if `versionedDeploy` is set. But I'll add a redundant check just in case to prevent errors.